styleguide/derek/global-components/anchor-scroll/anchor-scroll.js

```diff
@@ -0,0 +1,30 @@
+// This plugin can be used to anchor scroll to anything. Usage:
+// 1) Give your link the "rsAnchor-scroll" class
+// 2) Give the href of your link a hashed id (ie href="#someID")
+// 3) Set that id to the element you want scrolled to (ie id="someID")
+(($) => {
+  $.fn.anchorScroll = function anchorScroll() {
+    const plugin = this;
+    $(this).click(function scrollTo(e) {
+      e.preventDefault();
+      // we need to subtract the top nav or else scrolled section will look cut off.
+      const $this = $(this.hash);
+      const $top = $this.offset().top;
+      const dimensions = {
+        nav: $('.navbar-fixed-top').outerHeight(),
+        el: $this,
+        top: $top,
+      };
+      plugin.anchorScroll.caclTopScroll(dimensions);
+    });
+    $.fn.anchorScroll.caclTopScroll = (d) => {
+      if (typeof d.el.offset() !== typeof undefined && d.el.offset() !== false) {
+        $('html, body').animate({
+          scrollTop: (d.top - d.nav),
+        }, 500);
```

Comment:
Since this function is for "calculating" it should just do and return the calculated value.   So something like this: 

```
$.fn.anchorScroll.caclTopScroll = (top, nav) => {
  return top - nav;
}
```

Then just move the animate code to the area above in place of `plugin.anchorScroll.caclTopScroll(dimensions);`

```
$('html, body').animate({
  scrollTop: plugin.anchorScroll.calcTopScroll(d.top, d.nav),
}, 500);
```